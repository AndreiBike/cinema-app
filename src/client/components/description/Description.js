import React from 'react';
import headerPicture from '@root/client/images/headerPicture.jpg'
import Head from './head/head';
import Container from './container/Container';
import styles from './Description.module.css';
import cn from 'classnames';

import './Description.module.css';

const Description = (props) => {
  return (
<<<<<<< HEAD
    <div  className={styles.description}>
      <div className={styles.background}>
        <img src={headerPicture} alt="background image" />
=======
    <div className="description">
      <div className="description-background">
        <img src="./assets/headerPicture.jpg" alt="background image" />
>>>>>>> origin/develop
      </div>
      <Head />
      <Container movie = {props.movie}/>
    </div>
  )
} 

export default Description;
