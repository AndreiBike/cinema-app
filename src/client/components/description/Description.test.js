import React from 'react';
import { configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Description from './Description';

configure({ adapter: new Adapter() });

describe("<Description /> component testing", () => {

  let component = null;
  jest.mock('@root/client/store/store');

  it("test #1 should call componentDidMount once", () => {
    let componentDidMountSpy = spyOn(Description.prototype, 'componentDidMount');
    component = mount(<Description />);
    expect(componentDidMountSpy).toHaveBeenCalledTimes(1);
  });

  it("test #2 should populate the state", () => {
    component = mount(<Description />);
    setTimeout(()=>{
      expect(component.state().movie.id).toBe(2);
    }, 0);
  });

  it("+++capturing snapshot of Description ", ()=>{
    expect(component).toMatchSnapshot();
  })
})
