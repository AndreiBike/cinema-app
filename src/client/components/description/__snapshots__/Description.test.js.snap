// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description /> component testing +++capturing snapshot of Description  1`] = `ReactWrapper {}`;
